internal/controller/drclusters.go

```diff
@@ -245,6 +245,14 @@ func drClusterUndeploy(
 		return fmt.Errorf("drcluster '%v' referenced in one or more existing drPolicy resources", drcluster.Name)
 	}
 
+	if err := mwu.DeleteManifestWork(mwu.BuildManifestWorkName(util.MWTypeDRCConfig), drcluster.GetName()); err != nil {
+		return err
+	}
+
+	if _, err := mwu.FindManifestWork(mwu.BuildManifestWorkName(util.MWTypeDRCConfig), drcluster.GetName()); err == nil {
```

Comment:
@ShyamsundarR it's the only place we are checking manifest after deletion. If there is no error in line [248](https://github.com/RamenDR/ramen/pull/1485/files#diff-0fe31ec47a23104282e03d05496277a6352b98ffae3fcab8a4e8b7d24c9f534eR248:~:text=if%20err%20%3A%3D%20mwu.DeleteManifestWork(mwu.BuildManifestWorkName(util.MWTypeDRCConfig)%2C%20drcluster.GetName())%3B%20err%20!%3D%20nil%20%7B), the manifest will be deleted, so why we need an additional check?

Reply:
Good question, I want to ensure that the DRClusterConfig resource at the managed cluster is deleted before we proceed to delete the MW for the ramen components at the managed cluster. If we reach that before the resource is actually deleted, then the DRClusterConfig reconciler may not be existing to finalize the resource locally.

Hence we delete and wait till it is actually deleted, then we proceed with deletion of the ramen operators on the managed cluster.